### PR TITLE
[SEDONA-376] Add RS_SetValues

### PR DESCRIPTION
**Pull request by furqaankhan**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-376. The PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?

- Add RS_SetValues

## How was this patch tested?

- Passed added tests

## Did this PR include necessary documentation updates?

- Yes, I have updated the documentation update.

### CODE REVIEW

File: docs/api/sql/Raster-operators.md [hunk 1]
```diff
@@ -963,6 +963,47 @@ Output:
...
+
+!!!Note
+    If the mentioned `bandIndex` doesn't exist, this will throw an `IllegalArgumentException`.
+
+Spark SQL Example:
+
+```sql
+SELECT RS_BandAsArray(
+        RS_SetValues(
+            RS_AddBandAsArray(
```
Comment (jiayuasu) [inline docs/api/sql/Raster-operators.md:992]: This should be `RS_AddBandFromArray`

File: common/src/main/java/org/apache/sedona/common/raster/PixelFunctionEditors.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            }
+        }
+        return RasterUtils.create(wr, raster.getGridGeometry(), raster.getSampleDimensions());
+    }
+
+    public static GridCoverage2D setValues(GridCoverage2D raster, int band, int colX, int rowY, int width, int height, double[] values) {
+        return setValues(raster, band, colX, rowY, width, height, values, false);
+    }
+
+    private static void ensureBand(GridCoverage2D raster, int band) throws IllegalArgumentException {
```
Comment (jiayuasu) [inline common/src/main/java/org/apache/sedona/common/raster/PixelFunctionEditors.java:70]: Isn't this ensureBand function implemented at several places? If so, we need to share it across functions.

File: docs/api/sql/Raster-operators.md [hunk 3]
```diff
@@ -963,6 +963,47 @@ Output:
 -3.000000
 ```
 
+### RS_SetValues
+
+Introduction: Returns raster with new values for specified pixels for designated band.
```
Comment (jiayuasu) [inline docs/api/sql/Raster-operators.md:968]: Please clearly explain how this setValue works. It should cover the region specified by the min point from colX and rowY.

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: common/src/main/java/org/apache/sedona/common/raster/PixelFunctionEditors.java (hunk 2 above)
Comment (furqaankhan) [reply common/src/main/java/org/apache/sedona/common/raster/PixelFunctionEditors.java:70]: Sure! I will make one implementation public and use that.

File: common/src/main/java/org/apache/sedona/common/utils/RasterUtils.java [hunk 4]
```diff
@@ -284,6 +285,19 @@ public static int[] getGridCoordinatesFromWorld(GridCoverage2D raster, double lo
...
 
+    /***
+     * Throws an exception if band index is greater than the number of bands in a raster
+     * @param raster
+     * @param band
+     * @return
+     * @throws IllegalArgumentException
+     */
+    public static void ensureBand(GridCoverage2D raster, int band) throws IllegalArgumentException {
+        if (band > RasterAccessors.numBands(raster)) {
```
Comment (jiayuasu) [inline common/src/main/java/org/apache/sedona/common/utils/RasterUtils.java:296]: This should also check if the band is 1-indexed, e.g., band cannot `< 1`

File: common/src/main/java/org/apache/sedona/common/raster/PixelFunctionEditors.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+        Double noDataValue = null;
+        if (keepNoData) {
+            noDataValue = RasterBandAccessors.getBandNoDataValue(raster, band);
+        }
+        int iterator = 0;
+        for (int j = rowY; j < rowY + height; j++) {
+            for (int i = colX; i < colX + width; i++) {
+                double[] pixel = rasterCopied.getPixel(i, j, (double[]) null);
+                if (keepNoData && noDataValue != pixel[band - 1]) {
```
Comment (jiayuasu) [inline common/src/main/java/org/apache/sedona/common/raster/PixelFunctionEditors.java:53]: A better implementation of this if condition is 

```
if (keepNoData && noDataValue == pixel[band-1]) {
	continue;
}
else pixel[band-1] = values[iterator]

```

**Review by jiayuasu — CHANGES_REQUESTED** [review]

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**